### ISSUE TITLE
elemental cleanup 

### ISSUE THREAD

**Issue by pgonin** [rancher/elemental#476]
**Describe the solution you'd like:**
In a similar way rancher-cleanup removes all resources associated with Rancher to cleanup a cluster, there should be a way to cleanup elemental resources.
https://github.com/rancher/rancher-cleanup

This is in particular necessary in backup / restore operations that aim to return in a past point i tie on the same cluster.

**Comment by kkaempf** [comment]
This should probably be integrated into `rancher-cleanup` 🤔 

/cc @agracey

**Comment by davidcassany** [comment]
> This should probably be integrated into `rancher-cleanup` thinking

Indeed, if technically possible I'd say the best approach would be to have some sort of generic plugins cleanup. From a plugin perspective it feels not a very good solution that rancher utilities need to be aware of elemental (or any other plugin) specifics, rancher should not need to be aware of all plugins resources, as this slightly couples the releases of rancher plugins to rancher releases.

**Comment by agracey** [comment]
Agreed with @davidcassany about the coupling.

IMO, the extensions mechanism should allow for a way for an extension to specify the resources it needs for backup/restore. This would allow for the same decoupling but still get the central automation. 

@nwmac, is this something that's doable in the not-to-distant future?

**Comment by agracey** [comment]
I think my suggestion here is to allow the `UIPlugin.catalog.cattle.io` to specify CRDs provided and how backup/restore should work. Then the cleanup logic can be extended by the core team.

**Comment by pgonin** [comment]
Stale
Closing
